src/main/java/frc/robot/subsystems/Arm.java

```diff
@@ -0,0 +1,107 @@
+        sprocketRightMotor = new CANSparkMax(7, CANSparkMax.MotorType.kBrushless);
+        sprocketRightMotor.setInverted(true);  //Inverts the motor because one motor is on the other side
+        sprocketLeftEncoder = sprocketLeftMotor.getEncoder();
+        sprocketRightEncoder = sprocketRightMotor.getEncoder();
+        sprocketLeftPIDController = sprocketLeftMotor.getPIDController();
+        sprocketRightPIDController = sprocketRightMotor.getPIDController();
+        sprocketLeftPIDController.setP(Constants.armConstants.controllerGain);
+        sprocketRightPIDController.setP(Constants.armConstants.controllerGain); //declare
+        sprocketAbsoluteEncoder = new DutyCycleEncoder(0);
+    }
+    /*Sets the motor speed the desired value and if the speed input is positive, it checks
+     * the angle to make sure that the arm does not go above 90 degrees
+     */
+    public void setSprocketSpeed(double speed) {
+        double feedforwardcalculation = sprocketFeedforward.calculate(Math.toRadians(getSprocketAngle()),0.0);
+        sprocketLeftMotor.set(speed+feedforwardcalculation);
+        sprocketRightMotor.set(speed+feedforwardcalculation);
+        if(speed>0){
+        sprocketLimitStop();
+        }
+
+    }
+    /*
+     * Returns the current sprocket angle
+     */
+    public double getSprocketAngle() {
+        return (sprocketAbsoluteEncoder.getAbsolutePosition() * 360 + 20)  % 360 - 49.1;
+    }
+    /*
+     * Returns the current sprocket speed
+     */
+    public double getSprocketSpeed(){
+        return sprocketLeftEncoder.getVelocity();
+    }
+    /*
+     * Makes the sprocket full speed down
+     */
+    public void sprocketfullspeeddown(){
+        sprocketLeftMotor.set(-1);
+        sprocketRightMotor.set(-1);
+    }
+    /*
+     * Makes the sprocket full speed up
+     */
+    public void sprocketfullspeedup(){
+        sprocketLeftMotor.set(1);
+        sprocketRightMotor.set(1);
+    }
+    /*
+     * Moves the sprocket to the desired angle, and if the current sprocket angle
+     * is below -30 degrees, it divides the PID Calculation by 2 to ensure a soft approach
+     */
+    public void sprocketToPosition(double targetPos){
+        double SprocketPIDCalculations = pidController.calculate(getSprocketAngle(), targetPos);
+        double feedforwardcalculation = sprocketFeedforward.calculate(Math.toRadians(getSprocketAngle()),0.0);
+        if(getSprocketAngle() <= -30){
+            SprocketPIDCalculations /= 2;
+        }
+        double output = SprocketPIDCalculations + feedforwardcalculation;
+        sprocketLeftMotor.set(output);
+        sprocketRightMotor.set(output);
+    
+        sprocketLimitStop();
+        
+    }
+/* Stops the sprocket */
+    public void stopSprocket(){
+        double sprocketFeedforwardcalculation = sprocketFeedforward.calculate(Math.toRadians(getSprocketAngle()),0);
+        sprocketLeftMotor.set(-sprocketFeedforwardcalculation);
```

Comment:
I don't understand why the left motor is negative and the right motor positive.

Reply:
I think it's because the sprocket motors face opposite directions, but I don't know for sure.

Reply:
Since it's inverted this might not even work?